### ISSUE TITLE
index-wp-mysql-for-speed-update-filter.php has issues with tables that finish with $tablesToHandle postfix

### ISSUE THREAD

**Issue by EngHell** [OllieJones/index-wp-mysql-for-speeed#64]
I found the issue, while debugging an issue with the plugin Shortpixel image compressor, the plugin creates a table named `"{$wpdb->prefix}shortpixe_postmeta"` which gets caught by the code 
`$startsOK = 0 === substr_compare( $table, $prefix, 0, strlen( $prefix ) );`
and the code
`0 === substr_compare( $table, $tableToHandle, - strlen( $tableToHandle ) )`

A suggested fix would be to directly prepend prefix and the table to handle inside the foreach, and don't do both substr comparisons

edit:
related log
```
[27-Jul-2023 04:52:26 UTC] WordPress database error Table 'wp_harvestsite.wp_shortpixel_postmeta' doesn't exist for query SHOW CREATE TABLE wp_shortpixel_postmeta; /* From [harvest.org/wp-admin/plugins.php?action=activate&plugin=shortpixel-image-optimiser%2Fwp-shortpixel.php&plugin_status=all&paged=1&s&_wpnonce=d705190e8f] in [/nas/content/live/harvestsite/wp-content/mu-plugins/index-wp-mysql-for-speed-update-filter.php:101] */ made by activate_plugin, do_action('activate_shortpixel-image-optimiser/wp-shortpixel.php'), WP_Hook->do_action, WP_Hook->apply_filters, ShortPixel\Helper\InstallHelper::activatePlugin, ShortPixel\Helper\InstallHelper::checkTables, dbDelta, apply_filters('dbdelta_queries'), WP_Hook->apply_filters, index_mysql_for_speed_upgrade_filter
```

**Comment by OllieJones** [comment]
Thanks for the bug report. This will be fixed in 1.4.14

**Comment by EngHell** [comment]
Thanks, @OllieJones, I was reading my bug report and I noticed I could have provided some line numbers to improve it. I will be testing the update when I get a chance tomorrow.
